python/wikibot/wikibot.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Class for updating wikidata with extracted facts from a record file."""
+
+import pywikibot
+import sling
+import json
+import sys
+import datetime
+
+precision_map = {
+  sling.MILLENNIUM: pywikibot.WbTime.PRECISION['millenia'],
+  sling.CENTURY: pywikibot.WbTime.PRECISION['century'],
+  sling.DECADE: pywikibot.WbTime.PRECISION['decade'],
+  sling.YEAR: pywikibot.WbTime.PRECISION['year'],
+  sling.MONTH: pywikibot.WbTime.PRECISION['month'],
+  sling.DAY: pywikibot.WbTime.PRECISION['day']
+}
+
+class StoreFactsBot:
+  def __init__(self, test=True):
+    self.site = pywikibot.Site("wikidata", "wikidata")
+    self.repo = self.site.data_repository()
+
+    time_str = datetime.datetime.now().isoformat("-")[:19].replace(":","-")
+    if test:
+      record_file_name = "local/data/e/wikibot/test-birth-dates.rec"
+      time_str = "test-" + time_str
+    else:
+      record_file_name = "local/data/e/wikibot/birth-dates.rec"
+    status_file_name = "local/data/e/wikibot/wikibotlog-"+time_str+".rec"
+
+    try:
+      self.record_file = sling.RecordReader(record_file_name)
+      self.status_file = sling.RecordWriter(status_file_name)
+    except :
+      print "Can't open Sling record files", record_file_name, status_file_name
+      sys.exit()
+
+    self.store = sling.Store()
+    self.n_item = self.store["item"]
+    self.n_facts = self.store["facts"]
+    self.n_date_of_birth = self.store["P569"]
+    self.n_provenance = self.store["provenance"]
+    self.n_category = self.store["category"]
+    self.n_method = self.store["method"]
+    self.n_status = self.store["status"]
+    self.n_stored = self.store["stored"]
```

Comment:
The n_stored is really the wikidata revision. What about calling it "revision" instead to make this explicit?